### ISSUE TITLE
Coord can't have the same name as a model variable (xarray limitation)

### ISSUE THREAD

**Issue by michaelosthege** [arviz-devs/arviz#1642]
**Describe the bug**
This is a [known limitation of xarray](https://github.com/pydata/xarray/issues?q=is%3Aissue+and+the+same+name+as+one+of+its+dimensions).

I'm creating this issue to track it, and for others to find the information easier.

**To Reproduce**

```python
with pymc3.Model(coords={
    "lothar": ["first"]
}) as pmodel:
    pymc3.Lognormal("lothar", mu=-3, sd=0.8, dims=("lothar",))

with pmodel:
    trace = pm.sample(return_inferencedata=False)
    idata = arviz.from_pymc3(trace)
```

**Expected behavior**
ArviZ should be able to deal with random variables and their dimensions having the same name.

**Additional context**
As of today, I couldn't find a specific xarray issue where they track the problem.

However, [this comment on Stackoverflow](https://stackoverflow.com/questions/55109481/how-to-import-netcdf4-file-with-xarray-when-index-names-have-multiple-dimensions#comment96993085_55109481) says that NetCDF4 supports this, so it's really just a limitation with current xarray versions.

**Comment by OriolAbril** [comment]
This I am actually not sure it will be fixed at all on xarray side, from my understanding, the limitation comes from the fact that coordinates and variables are basically the same and are mostly separated for convenience. For example, coords can be accessed both as `dataset.coords["coord_name"]` _and_ `dataset["coord_name"]`, for variables to have the same name as coords, the 2nd can't work as it would have to return the variable as opposed to the coordinate. This could change in xarray 1.0, but I don't have enough insight as to venture a guess here. 

> ArviZ should be able to deal with random variables and their dimensions having the same name.

We may be able to do something like add underscores or `_dim` or something like that if we detect variables and coords with the same name in `dict_to_dataset`

**Comment by michaelosthege** [comment]
> We may be able to do something like add underscores or _dim or something like that if we detect variables and coords with the same name in dict_to_dataset

I thought about proposing `f"dim_{original}"` in the issue description, but then I realized that the users interact directly with the xarray objects.
We could, however, think about printing a `UserWarning` in PyMC3 even before starting the costly MCMCs.

It's clear that we need a workaround from the PyMC3/ArviZ side - it'd be best if we can agree on a best practice naming scheme..
* `f"_{original}"` → least character overhead
* `f"dim_{original}"` → similar to the automatic `dim_0`, `dim_1`... naming

Preferences?

**Comment by OriolAbril** [comment]
Our default is `{original}_dim_0`, `{original}_dim_1`... so I'd probably go with `{original}_dim` without number whenever this happens while printing a warning.

**Comment by michaelosthege** [comment]
> Our default is `{original}_dim_0`, `{original}_dim_1`... so I'd probably go with `{original}_dim` without number whenever this happens while printing a warning.

Right. That sounds good. I will adopt the `{original}_dim` pattern myself and will watch out for opportunities to warn/raise a `UserWarning` early on the PyMC3 side.
Then ArviZ can implement the automatic conversion + warning as a fallback against loosing precious samples.
